csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -82,32 +160,51 @@ private void CanRunInferenceOnAModel(uint graphOptimizationLevel, bool disableSe
                 // Run the inference
                 using (var results = session.Run(container))  // results is an IReadOnlyList<NamedOnnxValue> container
                 {
-                    Assert.Equal(1, results.Count);
+                    validateRunResults(results);
+                }
+
+                // Run Inference with RunOptions
+                using (var runOptions = new RunOptions())
+                {
+                    runOptions.LogTag = "CsharpTest";
+                    runOptions.Terminate = true;
+                    runOptions.LogVerbosityLevel = LogLevel.Error;
+                    IReadOnlyCollection<string> outputNames = session.OutputMetadata.Keys.ToList();
 
-                    float[] expectedOutput = LoadTensorFromFile(@"bench.expected_out");
-                    // validate the results
-                    foreach (var r in results)
+                    using (var results = session.Run(container, outputNames, runOptions))  // results is an IReadOnlyList<NamedOnnxValue> container
                     {
-                        Assert.Equal("softmaxout_1", r.Name);
+                        validateRunResults(results);
+                    }
```

Comment:
Does this run generate a log file that can be tested (i.e. it exists and has non-empty content)? If so, should we test for the log file?

Reply:
Currently all logs go to console, and we don't have run/session option to redirect the log output. Only logger override is available at the ENV, which is not exposed in C#. I don't know a good way to look into vstest console output either, and did not want to spend time parsing it.